### PR TITLE
feat: reimplement telemetry id

### PR DESCRIPTION
**Pull request by Jack-Works**
close MF-3679

The unique identifier based on the canvas fingerprint is introduced in #9012, currently, the unique identifier in Mask has three purposes:

- A/B test split
- API endpoint user-side offloading
- Identify users in Sentry

There are no privacy concerns for the first two uses, but for the third use, we should reconsider whether this implementation is appropriate (because this method is different from the first two, and the device identification will be uploaded as it is)

Canvas fingerprints are not friendly to user privacy protection because they are difficult to change.

Also, the current implementation is not available in Manifest V3.

This PR reimplements the telemetry id, now it is generated randomly and stored in the storage. It also open the ability to allow user to reset this track identifier.

### CODE REVIEW

File: packages/mask/src/setup.ui.0.ts [hunk 1]
```diff
@@ -1,9 +1,12 @@
-import type { KVStorageBackend } from '@masknet/shared-base'
+import { TelemetryID, type KVStorageBackend } from '@masknet/shared-base'
 import { setupMaskKVStorageBackend } from '../shared/kv-storage.js'
 import { setupLegacySettingsAtNonBackground } from '../shared/legacy-settings/createSettings.js'
 import Services from './extension/service.js'
 import { contentFetch } from './utils/fetcher.js'
+import { MaskMessages } from '../shared/messages.js'
 
+Services.Helper.getTelemetryID().then((id) => (TelemetryID.value = id))
```
Comment (guanbinrui) [inline packages/mask/src/setup.ui.0.ts:8]: Does it set as early enough? Before any `joinsABTest` is called.

File: packages/shared-base/tests/utils/ABTest.ts [hunk 2]
```diff
@@ -0,0 +1,15 @@
+import { describe, it, expect } from 'vitest'
+import { joinsABTest, getABTestSeed } from '../../src/utils/ABTest.js'
+
+describe('detect if an installation joins the A/B test', () => {
+    it('should calculate join A/B test correctly', () => {
+        expect(joinsABTest(100)).toBeTruthy()
+        expect(joinsABTest(0)).toBeFalsy()
```
Comment (UncleBill) [inline packages/shared-base/tests/utils/ABTest.ts:7]: Still have `1%` chance to be truthy.

Comment (UncleBill) [reply packages/shared-base/tests/utils/ABTest.ts:7]: Once failed https://github.com/DimensionDev/Maskbook/actions/runs/4977525342/jobs/8906690436?pr=9600

Comment (Jack-Works) [reply packages/shared-base/tests/utils/ABTest.ts:7]: oh, you're right. let me fix it in a new PR...

Comment (Jack-Works) [reply packages/shared-base/tests/utils/ABTest.ts:7]: already fixed!

**Pull request merged**